source/features/deep-reblame.tsx

```diff
@@ -0,0 +1,125 @@
+import features from '../libs/features';
+import loadingIcon from '../libs/icon-loading';
+import {getRepoGQL, getReference, looseParseInt} from '../libs/utils';
+
+const getPullRequestBlameCommit = mem(async (commit: string, prNumber: number): Promise<string | false> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "${commit}") {
+				... on Commit {
+					associatedPullRequests(last: 1) {
+						nodes {
+							number
+							mergeCommit {
+								oid
+							}
+							commits(last: 1) {
+								nodes {
+									commit {
+										oid
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+
+	const associatedPR = repository.object.associatedPullRequests.nodes[0];
+	const mergeCommit = associatedPR.mergeCommit.oid;
+
+	if (associatedPR.number === prNumber && mergeCommit === commit) {
+		return associatedPR.commits.nodes[0].commit.oid;
+	}
+
+	return false;
+});
+
+async function redirectToBlameCommit(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void> {
+	const blameLink = event.delegateTarget;
+	if (blameLink.href) {
+		if (event.altKey) {
+			event.preventDefault();
+		} else {
+			return; // Unmodified click on regular link: let it proceed
+		}
+	}
+
+	const blameHunk = blameLink.closest('.blame-hunk')!;
+	const prNumber = looseParseInt(select('.issue-link', blameHunk)!.textContent!);
+	const prCommit = select<HTMLAnchorElement>('a.message', blameHunk)!.href.split('/').pop()!;
+	const spinner = loadingIcon();
+	spinner.classList.add('mr-2');
+	blameLink.firstElementChild!.replaceWith(spinner);
+	// Hide tooltip after click, it’s shown on :focus
+	blurAccessibly(blameLink);
+
+	const prBlameCommit = await getPullRequestBlameCommit(prCommit, Number(prNumber));
+	if (prBlameCommit) {
+		const lineNumber = select('.js-line-number', blameHunk)!.textContent!;
+		blameLink.pathname = location.pathname.replace(getReference()!, prBlameCommit);
+		blameLink.hash = 'L' + lineNumber;
+		blameLink.click();
+		return;
+	}
+
+	// Restore the regular version link if there was one
+	if (blameLink.href) {
+		blameLink.setAttribute('aria-label', 'View blame prior to this change.');
+		blameLink.classList.remove('rgh-deep-blame');
+		spinner.replaceWith(versionIcon());
+	} else {
+		spinner.remove();
+	}
+
+	alert('The PR linked in the title didn’t create this commit');
+}
+
+function init(): void | false {
+	const pullRequests = select.all('[data-hovercard-type="pull_request"]');
+	if (pullRequests.length === 0) {
+		return false;
+	}
+
+	for (const pullRequest of pullRequests) {
+		const hunk = pullRequest.closest('.blame-hunk')!;
+
+		const reblameLink = select('.reblame-link', hunk)!;
+		if (reblameLink) {
+			reblameLink.setAttribute('aria-label', 'View blame prior to this change. Hold `Alt` to extract commits from this PR first');
+			reblameLink.classList.add('rgh-deep-blame');
+		} else {
+			select('.blob-reblame', hunk)!.append(
+				<a
```

Comment:
Plain links aren't focusable and the cursor doesn't change to `pointer`

Since we're using this link to navigate later (so we can't change it to the more appropriate `<buton type=button>`) I suggest
```suggestion
				<a
					href="#deep-reblame"
```

And then the condition on line 55 and 77 needs to be updated

Reply:
Can I do `href="#"` and then
	if href.length > 1

Reply:
No, `#` go to the top